ocaml/testsuite/tests/typing-unique/overwriting.ml

```diff
@@ -487,9 +487,70 @@ Uncaught exception: File "ocaml/parsing/location.ml", line 1106, characters 2-8:
 (*******************************)
 (* Overwriting inlined records *)
 
-(* CR uniqueness *)
+type constructor_update = Con of { x : string; y : string }
+[%%expect{|
+type constructor_update = Con of { x : string; y : string; }
+|}]
+
+let update = function
+  | (Con _ as c) ->
+    let x = overwrite_ c with Con { x = "foo" } in
+    x
+[%%expect{|
+Line 3, characters 12-47:
+3 |     let x = overwrite_ c with Con { x = "foo" } in
+                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alert Translcore: Overwrite not implemented.
+Uncaught exception: File "ocaml/parsing/location.ml", line 1106, characters 2-8: Assertion failed
+
+|}]
 
-(******************************)
-(* Overwriting mutable fields *)
+let update = function
+  | (Con c1 as c) ->
+    let x = overwrite_ c with Con { c1 with x = "foo" } in
+    x
+[%%expect{|
+Line 3, characters 12-55:
+3 |     let x = overwrite_ c with Con { c1 with x = "foo" } in
+                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alert Translcore: Overwrite not implemented.
+Uncaught exception: File "ocaml/parsing/location.ml", line 1106, characters 2-8: Assertion failed
 
-(* CR uniqueness *)
+|}]
+
+let update = function
+  | (Con c) ->
+    let x = overwrite_ c with { x = "foo" } in
+    x
```

Comment:
It gives the same output (with the new variable in the error obviously)

Reply:
But I think that should probably be accepted. As usual, it's OK if it's not, but wrapping the result in `Con` means that the type of the inlined record does not actually escape.

Reply:
Ah, but this is tricky because the compiler never allows binding inlined records to variables. For example:

```
let update = function
  | (Con c) -> let x = c in Con x
[%%expect{|
Line 2, characters 23-24:
2 |   | (Con c) -> let x = c in Con x
                           ^
Error: This form is not allowed as the type of the inlined record could escape.
|}]
```

So this would be a larger change in the compiler that actually is not specific to overwriting.